dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+'''
+'''
+
+from __future__ import absolute_import
+
+# standard imports
+import logging
+import uuid
+import time
+import os
+import json
+from datetime import datetime
+
+# internal imports
+from dripline import core
+from dragonfly.implementations import EthernetProvider
+
+
+__all__ = []
+
+logger = logging.getLogger(__name__)
+
+__all__.append('PsyllidProvider')
+class PsyllidProvider(core.Provider, core.Spime):
+    '''
+    Provider for direct communication with Psyllid 
+    '''
+    def __init__(self,
+                 psyllid_queue='psyllid',
+                 set_condition_list = [],
+                 **kwargs):
+
+	core.Provider.__init__(self, **kwargs)        
+        self.psyllid_queue = psyllid_queue
+        self._set_condition_list = set_condition_list
+        self.status = None
+        self.status_value = None
+               
+        self.channel_dict = {'a': 0, 'b': 1, 'c': 2}
+        self.freq_dict = {'a': None, 'b': None, 'c': None}
+        self.mode = None
+        
+        
+    def _finish_configure(self):
+        try:
+            if self.request_status()==0:
+                self.activate()
+        except:
+            raise core.exceptions.DriplineGenericDAQError('Psyllid instance with this queue is not responding')
+
+        self.mode = 'streaming'
+        try:
+            self.set_default_central_frequencies()
+        except:
+            self.mode = 'triggered'
+            self.set_default_central_frequencies()
+        self.get_number_of_streams()
+        
+        
+    def set_default_central_frequencies(self):
+        freq_dict = {'a': 800e6, 'b': 800e6, 'c': 800e6}
+        self.set_all_central_frequencies(freq_dict)
+        
+
+    def request_status(self):
+        logger.info('Checking Psyllid status')
+        
+        try:
+            result = self.provider.get(self.psyllid_queue+'.daq-status', timeout=10)
+            self.status = result['server']['status']
+            self.status_value = result['server']['status-value']
+            logger.info('Psyllid is running. Status is {}'.format(self.status))
+            logger.info('Status in numbers: {}'.format(self.status_value))
+            return self.status_value
+
+        except:
+            logger.warning('Psyllid is not running or sth. else is wrong')
+            self.status=None
+            self.status_value=None
+            logger.info('Status is {}'.format(self.status))
+            return self.status_value
```

Comment:
Is this return something send back in a Reply? Status (success/failure) information needs to be in a return code, not a payload value. If "Status" means 'State' (as in "activated" or "idle" or whatever, where 'State' is some configurable parameter then maybe this is okay, but returning it in an except block seems questionable.